.github/workflows/plugin-packages.yml

```diff
@@ -190,6 +196,51 @@ jobs:
     - name: Test Preflight Plugin Locally
       run: make test-preflight-plugin-locally
 
+  # build_push_preflight_image job builds preflight docker image, and pushes it to public docker registry,
+  # if changes for tags('/ref/tags') are pushed or otherwise pushes to private docker registry
+  build_push_preflight_image:
+    runs-on: ubuntu-latest
+    needs: [pre-build, build_test_preflight]
+    # for tags('ref/tags/'), 'needs.pre-build.outputs.preflight' will be 'false' so to run 'build_test_preflight' job on tags as well,
+    # added condition 'startsWith(github.ref, 'refs/tags/')'
+    if: (needs.pre-build.outputs.preflight == 'true') || startsWith(github.ref, 'refs/tags/')
+    env:
+      GCR_PRIVATE_DOCKER_REGISTRY: "gcr.io"
+      GCR_PUBLIC_DOCKER_REGISTRY: "eu.gcr.io"
+      GCR_PROJECT: ${{ secrets.GCS_PROJECT_ID }}
+      IMAGE_NAME: "preflight"
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Setup gcloud CLI
+      - uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCS_PROJECT_ID }}
+          service_account_key: ${{ secrets.GKE_SA_KEY }}
+          export_default_credentials: true
+
+      # Configure docker to use the gcloud command-line tool as a credential helper
+      - name: Set up docker auth
+        run: |
+          gcloud auth configure-docker
+
+      # This step is run when push is not for a tag
+      - name: Sets image tag env to private registry for internal build
+        if: (needs.pre-build.outputs.testPreflight == 'true') && !startsWith(github.ref, 'refs/tags/')
```

Comment:
does `!startsWith` syntax work? For earlier versions, this exact syntax was not supported.

Reply:
yes. tried this on the current workflow as well.